mender-auth/client/client_test.cpp

```diff
@@ -0,0 +1,117 @@
+// Copyright 2023 Northern.tech AS
+//
+//    Licensed under the Apache License, Version 2.0 (the "License");
+//    you may not use this file except in compliance with the License.
+//    You may obtain a copy of the License at
+//
+//        http://www.apache.org/licenses/LICENSE-2.0
+//
+//    Unless required by applicable law or agreed to in writing, software
+//    distributed under the License is distributed on an "AS IS" BASIS,
+//    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//    See the License for the specific language governing permissions and
+//    limitations under the License.
+
+#include <mender-auth/client/client.hpp>
+
+#include <string>
+#include <iostream>
+
+#include <gmock/gmock.h>
+#include <gtest/gtest.h>
+
+#include <common/testing.hpp>
+#include <common/http.hpp>
+#include <common/io.hpp>
+
+using namespace std;
+
+namespace error = mender::common::error;
+namespace http = mender::http;
+namespace io = mender::common::io;
+
+using TestEventLoop = mender::common::testing::TestEventLoop;
+
+const string TEST_PORT = "8088";
+
+class AuthClientTests : public testing::Test {
+protected:
+	const char *test_device_identity_script = "./mender-device-identity";
```

Comment:
I think we should try to stick to using `TemporaryDirectory` for test files. It removes the need for cleanup code, and also takes us one step closer to `ctest -j` working (though it still doesn't currently, due to the reused ports in various places).

All it needs is to be a member, and then use `tmpdir.Path()` here.

Reply:
Is that how tmpdir works :laughing: